src/codegen/compile.ml

```diff
@@ -9376,6 +9402,48 @@ and compile_prim_invocation (env : E.t) ae p es at =
     SR.Vanilla,
     GC.get_collector_instructions env ^^ BigNum.from_word64 env
 
+  (* Regions *)
+
+  | OtherPrim "regionNew", [] ->
+    SR.Vanilla,
+    Region.new_ env
+
+  | OtherPrim "regionId", [e0] ->
+    SR.UnboxedWord32,
```

Comment:
This telling the compiler that the value is an unboxed word32, not a Motoko Vanilla Value, so it will coerce it by shifting left 1 bit, hence 2 becomes 4. I think.

If Region.id is already returning a Motoko value, change the stack representation to SR.Vanilla.